net-libs/grpc/grpc-1.12.1.ebuild

```diff
@@ -0,0 +1,154 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 python3_{4,5,6} )
+DISTUTILS_OPTIONAL=1
+
+inherit distutils-r1 flag-o-matic toolchain-funcs
+
+PROTOBUF_VERSION="3.5.2"
+
+DESCRIPTION="Modern open source high performance RPC framework"
+HOMEPAGE="http://www.grpc.io"
+SRC_URI="
+	https://github.com/${PN}/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz
+	tools? ( https://github.com/google/protobuf/archive/v${PROTOBUF_VERSION}.tar.gz -> protobuf-${PROTOBUF_VERSION}.tar.gz )
+"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="examples doc python systemtap tools"
+
+REQUIRED_USE="
+	python? ( ${PYTHON_REQUIRED_USE} )
+	tools? ( python )
+"
+
+DEPEND="
+	>=dev-libs/openssl-1.0.2:0=[-bindist]
+	>=dev-libs/protobuf-3.5:=
+	dev-util/google-perftools
+	net-dns/c-ares:=
+	sys-libs/zlib:=
+	python? ( ${PYTHON_DEPS}
+		dev-python/coverage[${PYTHON_USEDEP}]
+		dev-python/cython[${PYTHON_USEDEP}]
+		>=dev-python/protobuf-python-3.5.1:=[${PYTHON_USEDEP}]
+		dev-python/six[${PYTHON_USEDEP}]
+		dev-python/wheel[${PYTHON_USEDEP}]
+		virtual/python-enum34[${PYTHON_USEDEP}]
+		virtual/python-futures[${PYTHON_USEDEP}]
+		doc? (
+			dev-python/sphinx[${PYTHON_USEDEP}]
+			dev-python/sphinx_rtd_theme[${PYTHON_USEDEP}]
+		)
+	)
+	systemtap? ( dev-util/systemtap )
+"
+
+RDEPEND="${DEPEND}"
+
+PATCHES=(
+	"${FILESDIR}/0001-grpc-1.11.0-Fix-cross-compiling.patch"
+	"${FILESDIR}/0002-grpc-1.3.0-Fix-unsecure-.pc-files.patch"
+	"${FILESDIR}/0003-grpc-1.3.0-Don-t-run-ldconfig.patch"
+	"${FILESDIR}/0004-grpc-1.11.0-fix-cpp-so-version.patch"
+	"${FILESDIR}/0005-grpc-1.11.0-pkgconfig-libdir.patch"
+	"${FILESDIR}/0006-grpc-1.12.1-allow-system-openssl.patch"
+	"${FILESDIR}/0007-grpc-1.12.1-allow-system-zlib.patch"
+	"${FILESDIR}/0008-grpc-1.12.1-allow-system-cares.patch"
+)
+
+src_prepare() {
+	sed -i 's@$(prefix)/lib@$(prefix)/$(INSTALL_LIBDIR)@g' Makefile || die
+	default
+	use python && distutils-r1_src_prepare
+}
+
+python_prepare() {
+	if use tools; then
+		rm -r third_party/protobuf || die
+		ln -s "${S}"/../protobuf-"${PROTOBUF_VERSION}" third_party/protobuf || die
+
+		pushd tools/distrib/python/grpcio_tools >/dev/null || die
+			ln -s ../../../../src || die
+			ln -s ../../../../third_party || die
+			ln -s ../../../../ grpc_root || die
```

Comment:
Don't indent these one more level.
also, looks like there is a space between the "../" and "grpc_root", is that intentional? if it is maybe putting "./grpc_root" would make it clearer in future. or doing two args for the higher ones (eg ln -s ..../src ./) then they match up and we wont accidentally mix it up in the future.

Reply:
Indent here just makes code more readable, at least for me. It's clear that I do stuff in non-default dir.  I'll unindent or maybe separate by empty lines. that's just personal preference.

yes, space before grpc_root is intentional. it links basedir here. and compiles with `-Igrpc_root` later.

I'll see what I can do to make it more readable and safe.

 probably specifying target is better, e.g.
```
ln -s ../../src ./src
ln -s ../../ ./grpc_root
```